packages/lang/llvm/package.mk

```diff
@@ -48,15 +48,27 @@ pre_configure() {
 }
 
 pre_configure_host() {
-  case "${TARGET_ARCH}" in
+  case "${MACHINE_HARDWARE_NAME}" in
+    "aarch64")
+      LLVM_BUILD_TARGETS="AArch64"
+      ;;
     "arm")
-      LLVM_BUILD_TARGETS="X86\;ARM"
+      LLVM_BUILD_TARGETS="ARM"
+      ;;
+    "x86_64")
+      LLVM_BUILD_TARGETS="X86"
       ;;
+  esac
+
+  case "${TARGET_ARCH}" in
     "aarch64")
-      LLVM_BUILD_TARGETS="X86\;AArch64"
+      LLVM_BUILD_TARGETS+="\;AArch64"
+      ;;
+    "arm")
+      LLVM_BUILD_TARGETS+="\;ARM"
       ;;
     "x86_64")
-      LLVM_BUILD_TARGETS="X86\;AMDGPU"
+      LLVM_BUILD_TARGETS+="\;X86\;AMDGPU"
```

Comment:
Typo here: X86 is already brought from above?

Reply:
> Typo here: X86 is already brought from above?

Yeah I thought about that logic too (note it works having X86\;X86\;AMDGPU). Happy for a better logic....

On an X86 Build Host we need:
- X86\;ARM
- X86\;AArch64
- X86\;AMDGPU or X86\;X86\;AMDGPU works

On a AArch64 build host we need:
- AArch64\;ARM
- AArch64 or AArch64\;AArch64 works
- AArch64\;X86\;AMDGPU

and on ARM
- ARM or ARM\;ARM
- ARM or ARM\;AArch64 works
- ARM\;X86\;AMDGPU

Reply:
Ok then. It looks confusing because template is different between targets :(